engine/src/execution_graph/logic_controllers/BatchAggregationProcessing.cpp

```diff
@@ -239,81 +261,86 @@ MergeAggregateKernel::MergeAggregateKernel(std::size_t kernel_id, const std::str
+        std::tie(group_column_indices, aggregation_input_expressions, aggregation_types,
+            aggregation_column_assigned_aliases) = ral::operators::parseGroupByExpression(this->expression,concatenated->num_columns());
+
+        std::vector<int> mod_group_column_indices;
+        std::vector<std::string> mod_aggregation_input_expressions, mod_aggregation_column_assigned_aliases, merging_column_names;
+        std::vector<AggregateKind> mod_aggregation_types;
+        std::tie(mod_group_column_indices, mod_aggregation_input_expressions, mod_aggregation_types,
+            mod_aggregation_column_assigned_aliases) = ral::operators::modGroupByParametersForMerge(
+            group_column_indices, aggregation_types, concatenated->names());
+
+        std::unique_ptr<ral::frame::BlazingTable> columns = nullptr;
+        if(aggregation_types.size() == 0) {
+            columns = ral::operators::compute_groupby_without_aggregations(
+                    concatenated->toBlazingTableView(), mod_group_column_indices);
+        } else if (group_column_indices.size() == 0) {
+            // aggregations without groupby are only merged on the master node
+            if( context->isMasterNode(ral::communication::CommunicationData::getInstance().getSelfNode()) ) {
+                columns = ral::operators::compute_aggregations_without_groupby(
+                        concatenated->toBlazingTableView(), mod_aggregation_input_expressions, mod_aggregation_types,
+                        mod_aggregation_column_assigned_aliases);
+            } else {
+                // with aggregations without groupby the distribution phase should deposit an empty dataframe with the right schema into the cache, which is then output here
+                columns = std::move(concatenated);
+            }
         } else {
-            // with aggregations without groupby the distribution phase should deposit an empty dataframe with the right schema into the cache, which is then output here
-            columns = std::move(concatenated);
+            columns = ral::operators::compute_aggregations_with_groupby(
+                    concatenated->toBlazingTableView(), mod_aggregation_input_expressions, mod_aggregation_types,
+                    mod_aggregation_column_assigned_aliases, mod_group_column_indices);
         }
-    } else {
-        columns = ral::operators::compute_aggregations_with_groupby(
-                concatenated->toBlazingTableView(), mod_aggregation_input_expressions, mod_aggregation_types,
-                mod_aggregation_column_assigned_aliases, mod_group_column_indices);
+        eventTimer.stop();
+
+        auto log_output_num_rows = columns->num_rows();
+        auto log_output_num_bytes = columns->sizeInBytes();
+
+        // events_logger->info("{ral_id}|{query_id}|{kernel_id}|{input_num_rows}|{input_num_bytes}|{output_num_rows}|{output_num_bytes}|{event_type}|{timestamp_begin}|{timestamp_end}",
```

Comment:
why was this commented out? should we just delete it?

Reply:
That code actually works fine. Uncommented.